### PR TITLE
EVG-16117: Add Git Tag field

### PR DESCRIPTION
**Pull request by sophstad**
EVG-16117

### Description 
- Add git tag field
- Edit general alias implementation to make it more modular
  - Remove AliasRow component
  - Allow any array form to use expandable cards (not tied to aliases)
- Improve UX so that toggling between regex/tag input does not clear the input

### Screenshots
image

### Testing 
- Added unit tests
- Updated Cypress tests

### CODE REVIEW

**Comment by ablack12** [comment]
I tried saving this page with defined git tag aliases and received an error code.
image

**Review by ablack12 — APPROVED** [review]
I imagine this isn't related to your change, and I'm not sure it's something we can every fix (or at least not as part of the POC 😄 ) but I noticed that when I type in the text box it takes a full beat for what I type to become visible; not sure if this is a debounce thing or etc but wanted to call it out.

**Comment by sophstad** [comment]
> I imagine this isn't related to your change, and I'm not sure it's something we can every fix (or at least not as part of the POC 😄 ) but I noticed that when I type in the text box it takes a full beat for what I type to become visible; not sure if this is a debounce thing or etc but wanted to call it out.

Yeah, I think this is less of a debounce issue and more of a "there's a lot of stuff on this page" problem 😢. I made EVG-16512 to look into it

File: cypress/integration/projectSettings/project_settings.ts [hunk 1]
```diff
@@ -268,6 +264,32 @@ describe("Project Settings when not defaulting to repo", () => {
...
+
+    it("Allows adding a git tag alias", () => {
+      cy.dataCy("git-tag-enabled-radio-box").children().first().click();
+      cy.dataCy("add-button").contains("Add Git Tag").parent().click();
+      cy.dataCy("git-tag-input").type("myGitTag");
+      cy.dataCy("remote-path-input").type("./evergreen.yml");
+    });
+
+    it("Updates the Require Signed field", () => {
+      cy.dataCy("require-signed-radio-box").children().first().click();
```
Comment (khelif96) [inline cypress/integration/projectSettings/project_settings.ts:281]: There is no assertion here.

File: src/components/SpruceForm/LeafyGreenWidgets.tsx [hunk 2]
```diff
@@ -299,6 +303,51 @@ export const LeafyGreenTextArea: React.FC<WidgetProps> = ({
...
+  value,
+}) => {
+  const {
+    "aria-controls": ariaControls,
+    "data-cy": dataCy,
+    enumOptions,
+  } = options;
+
+  if (!Array.isArray(enumOptions)) {
+    console.error("Non-Array passed into Segmented Control");
```
Comment (khelif96) [inline src/components/SpruceForm/LeafyGreenWidgets.tsx:320]: We should log this to bugsnag instead using `reportError`.

File: src/components/SpruceForm/LeafyGreenWidgets.tsx [hunk 3]
```diff
@@ -299,6 +303,51 @@ export const LeafyGreenTextArea: React.FC<WidgetProps> = ({
...
+    return null;
+  }
+
+  const idPrefix = id.substring(0, id.lastIndexOf("_"));
```
Comment (khelif96) [inline src/components/SpruceForm/LeafyGreenWidgets.tsx:324]: i'm curious about this transformation?  Why is this necessary?

Comment (sophstad) [reply src/components/SpruceForm/LeafyGreenWidgets.tsx]: Good point, I was experimenting with programmatically computing the IDs that the segmented control controls, but it's way more straightforward to specify them manually. Removed!

**Review by khelif96 — APPROVED** [review]

**Pull request merged**